presto-google-sheets/src/main/java/com/facebook/presto/google/sheets/SheetsConfig.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.google.sheets;
+
+import com.facebook.airlift.configuration.Config;
+import com.facebook.airlift.configuration.ConfigDescription;
+import io.airlift.units.Duration;
+import io.airlift.units.MinDuration;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import java.util.concurrent.TimeUnit;
+
+public class SheetsConfig
+{
+    private String credentialsFilePath;
+    private String metadataSheetId;
+    private int sheetsDataMaxCacheSize = 1000;
```

Comment:
A quick question: What's the difference between the `sheetsDataMaxCacheSize` here and the 10,000 used in this line `String defaultRange = "$1:$10000";` in the `SheetsConfig.java`? Are they referring to the same value?

Reply:
This is actually an internal optimization to this connector. This connector stores table information in a metadata sheet, so for every query it needs to make calls to metadata sheet to lookup. But we can cache this information locally, so it doesn't have to query remote metadata sheet every time. And this 1000 specifies that the local cache can store 1000 table info in it.

Regarding the "$1:$10000" it is the range in a google sheet (a google sheet notion).

Reply:
Thanks for your detailed clarification!